File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/StarWarsApplication.java
package com.dani_chuks.andeladeveloper.starwars;

import android.app.Activity;
import android.app.Application;

import com.crashlytics.android.Crashlytics;
import com.dani_chuks.andeladeveloper.starwars.dagger.DaggerAppComponent;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;
import io.fabric.sdk.android.Fabric;

public class StarWarsApplication extends Application implements HasActivityInjector {

    @Inject
    DispatchingAndroidInjector<Activity> activityDispatchingAndroidInjector;

    @Override
    public void onCreate() {
        super.onCreate();
        Fabric.with(this, new Crashlytics());
        DaggerAppComponent.builder()
                .application(this)
                .build()
                .inject(this);

    }

    @Override
    public AndroidInjector<Activity> activityInjector() {
        return activityDispatchingAndroidInjector;
    }
}

File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/StarWarsApplication.kt
package com.dani_chuks.andeladeveloper.starwars

import android.app.Activity
import android.app.Application
import com.dani_chuks.andeladeveloper.starwars.di.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasAndroidInjector
import javax.inject.Inject

class StarWarsApplication : Application(), HasAndroidInjector {

    @Inject
    lateinit var activityDispatchingAndroidInjector: DispatchingAndroidInjector<Any>

    override fun onCreate() {
        super.onCreate()
        //        Fabric.with(this, new Crashlytics());
        DaggerAppComponent.builder()
                .application(this)
                .build()
                .inject(this)

    }

    override fun androidInjector(): AndroidInjector<Any> {
        return activityDispatchingAndroidInjector
    }
}